src/server/ApiSorter.cpp

```diff
@@ -168,14 +168,25 @@ PCONSOLE_API_MSG ApiSorter::ConsoleDispatchRequest(_Inout_ PCONSOLE_API_MSG Mess
     Message->State.WriteOffset = Message->msgHeader.ApiDescriptorSize;
     Message->State.ReadOffset = Message->msgHeader.ApiDescriptorSize + sizeof(CONSOLE_MSG_HEADER);
 
+    HRESULT hr = S_OK;
+    try
+    {
+        hr = (*Descriptor->Routine)(Message, &ReplyPending);
+    }
+    catch (const wil::ResultException& e)
+    {
+        hr = e.GetStatusCode();
+    }

```

Comment:
this seems quite different from what we have today - does it deserve its own PR?

Reply:
Ironically it's what prompted me to make this change before I drifted off. I'd be happy to split this into its own PR.